src/MTGCore/Services/SearchFilterMapper.cs

```diff
@@ -0,0 +1,60 @@
+﻿using MTGCore.Services.Exceptions;
+using MTGCore.ViewModels;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace MTGCore.Services
+{
+    public class SearchFilterMapper : ISearchFilterMapper
+    {
+        private Dictionary<string, string> filterParameters = new Dictionary<string, string>();
+
+        private Dictionary<Type, Delegate> typeProcessorMap = new Dictionary<Type, Delegate>() 
```

Comment:
Should use the `readonly` keyword. There may be some logic in making it a `ReadOnlyDictionary`.

Also, `typeProcessorMap` could use the `IDictionary/IReadonlyDictionary` type.